### ISSUE TITLE
PHP artisan install returns "No application encryption key has been specified"

### ISSUE THREAD

**Issue by KodySalak** [rconfig/rconfig#71]
**Describe the bug**
After following the steps to install over 5 times line by line, I still come up with a "No application encryption key has been specified." error when running step 9 of the install guide on the main page of this repo.

**To Reproduce**
1. Follow the [Ubuntu setup script](https://docs.rconfig.com/getstarted/os-setup/#ubuntu-1804-scripts) instructions to set up prereqs
2. make all the necessary changes (apache, php installs, supervisor, .env file copying and modifying, etc.)
3. php artisan install
4. fails with above error.

**Expected behavior**
Install like it's supposed to. 

**Screenshots**
![error](https://github.com/rconfig/rconfig/assets/7674269/7890906c-ccb0-4b31-be51-99d804d99f39)

**rConfig Version Information (You can copy this from the rConfig dashboard):**
OS VERSION
Ubuntu Server 20.04 Focal Fossa
PHP / LARAVEL VERSION
PHP 8.2.10 / _running `php artisan --version` returns the same error as above_
REDIS VERSION
5.0.7
MYSQL VERSION
8.0.34-0ubuntu0.20.04.1

**Logs info (can be retrieved from /var/www/html/rconfig6/storage/logs - try to paste relevant logs only)
There are no logs available. 

**Additional context**
Like I've mentioned before, I've run this around 5 times now with no change. I cannot get it to deploy!
I've tried to run `php artisan key:generate` to get a key into the ENV file, but it just wont work (errors with same error). 
The current permissions for the entire rconfig directory is root:root
The MySQL database is on another server, and I am trying to connect to an old database that is currently running a CentOS box with rConfig 6 installed - I need to move off of this box to Ubuntu. 

Any help would be GREATLY appreciated.

**Comment by stephenstack** [comment]
Sorry you had this issue. It has been reported a few times, but I have not been able to find a root cause. One thing that I am curios about is edits to the .env file while setting up the DB before doing the key:generating. I'm curious if any formatting or whitespace issues in the file are causing the issue. 

Try this;

Copy .env.example to .env:
cp -a .env.example .env

Update the DB values. The DB values should not have any spaces, and can be enclosed in quotes.

Generate a key:
php artisan key:generate

**Comment by KodySalak** [comment]
Thanks for the quick reply.
was in the middle of typing a response when you edited - so I did both generate before I changed the DB info and after

Before I changed the DB info:
 I got the following 3 errors, but these should be expected because the name of the database host is DBNAME... correct? 

`SQLSTATE[HY000] [2002] php_network_getaddresses: getaddrinfo for DBHOST failed: Temporary failure in name resolution (Connection: mysql, SQL: select * from information_schema.tables where table_schema = DBNAME and table_name = settings and table_type = 'BASE TABLE')`

`SQLSTATE[HY000] [2002] php_network_getaddresses: getaddrinfo for DBHOST failed: Temporary failure in name resolution`

`PDO::__construct(): php_network_getaddresses: getaddrinfo for DBHOST failed: Temporary failure in name resolution`

I looked in the .env and there still wasn't a key generated.

After the DB info was changed: 
I received the same "No application encryption key has been specified." error.

My config file as it sits after just changing the DB info:
```
APP_NAME="rConfig6 - Network Configuration Management"
APP_ENV=production
APP_KEY=
APP_DEBUG=false
APP_URL="https://SERVER.DOMAIN.LOCAL"
APP_DIR_PATH=/var/www/html/rconfig

LOG_CHANNEL=daily
LOG_LEVEL=warning

TEST_DB_HOST=
TEST_DB_DATABASE=
TEST_DB_USERNAME=
TEST_DB_PASSWORD=

DB_CONNECTION=mysql
DB_HOST="mysql01.mydomain.local"
DB_PORT=3306
DB_DATABASE="rConfig"
DB_USERNAME="rconfigsa"
DB_PASSWORD="32CharacterPasswordWithNoSymbols"
DB_STORAGE_LOCATION=/storage/app/rconfig/mysql

BROADCAST_DRIVER=log
CACHE_DRIVER=redis
CACHE_PREFIX=rConfig6Core
HORIZON_PREFIX=rConfig6Core
HORIZON_LOCAL_TIMEOUT=60
HORIZON_PROD_TIMEOUT=60
QUEUE_CONNECTION=redis
QUEUE_DRIVER=redis
SESSION_DRIVER=redis
SESSION_LIFETIME=120

#DOCKER REDIS ENV CONFIG
REDIS_HOST_DOCKER=rconfig-redis
REDIS_HOST=localhost
REDIS_PASSWORD=null
REDIS_PORT=6379

#DOCKER EXPOSED PORTS
EXPOSED_APP_PORT=8080
EXPOSED_DB_PORT=3307
EXPOSED_HORIZON_PORT=8081
EXPOSED_REDIS_PORT=7000

MAIL_DRIVER=smtp
MAIL_HOST=
MAIL_PORT=
MAIL_USERNAME=null
MAIL_PASSWORD=null
MAIL_ENCRYPTION=null
MAIL_FROM_ADDRESS=null
MAIL_FROM_NAME=null

RCONFIG_API_TOKEN=V6Core
RCONFIG_SUB_ID="V6Core"
RCONFIG_SUB_NAME="V6Core"
RCONFIG_SUB_STATUS="Active"
RCONFIG_SUB_EXPIRY="N/A"

TIMEZONE=Europe/Dublin

ENLIGHTN_SKIP_ENVIRONMENT_SPECIFIC=false
```

**Comment by stephenstack** [comment]
Thanks, this needs a live review. Please make contact, so we can investigate it. 

https://www.rconfig.com/#contact

**Comment by KodySalak** [comment]
Can do - expect something momentarily.

**Comment by rconfig** [comment]
closed with #72  which defines the correct steps to update rConfig. Never use the php artisan install script to update existing installations, this is only for new installations. If you do, the script with overwrite the existing 'APP_KEY' in the .env file, which is the encryption key for the app. Then you get errors like 'encryption key not found' or 'MAC is invalid
